fof/View/DataView/Json.php

```diff
@@ -281,6 +281,16 @@ protected function onBeforeRead($tpl = null)
 	}
 
 	/**
+	 * The event which runs when we are displaying a single item JSON view
+	 *
+	 * @param   string  $tpl  The view sub-template to use
+	 */
+	protected function onAfterSave($tpl = null)
+	{
+		self::onBeforeRead($tpl);
```

Comment:
Is this correct? You are calling on **before** read in the **after** save. Is there a reason?